presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -535,13 +543,21 @@ private void addToHiveSplitSourceRoundRobin(List<Iterator<HiveSplit>> iteratorLi
             Map<Integer, HiveType> columnCoercions)
             throws IOException
     {
+        Path filePath = new Path(path);
```

Comment:
The more general truth is that the index is coming with lzop files. It's no aware of the data format. That's the reason why we want to contribute the patch for splittable lzo to upstream since they have support for lzo text and they should be able to benefit with our patch